API.md

```diff
@@ -182,7 +182,7 @@ Returns router middleware which dispatches a route matching the request.
 <a name="module_koa-router--Router+use"></a>
 
 #### router.use([path], middleware) ⇒ <code>Router</code>
-Use given middleware.
+Use given middleware, __if and only if__, a route is matched. Context: [bug report](https://github.com/ZijianHe/koa-router/issues/182) & [test](https://github.com/koajs/router/blob/2e69eca36739171ee23dde9f616a133488378a62/test/lib/router.js#L197-L222)
```

Comment:
```suggestion
Use given middleware, __if and only if__, a route is matched.
```

I believe the links not needed here and should be removed to match the style of the rest of the document.

Reply:
Done, added. The test link for ref in the commit message just in case someone wants to know why this was added.